### PR TITLE
#883: Update packages

### PR DESCRIPTION
**Pull request by tomuben**
related to https://github.com/exasol/script-languages-release/issues/883

1. Update `pandas`, `numpy` and `pyarrow` in template-Python3.10 flavor.
2. Deactivate conversion tests for pandas dataframes for  `halffloat[pyarrow]` + `float and float16[pyarrow] and float` because in newer versions of pandas this conversion is fixed, but we cannot (yet) use the latest versions in all flavors (`pandas` 2.1 and later require >= Python3.9)
3. Update ignore list in import_modules.py (due to latest packages in new `standard-all` flavor)
4. Changed test `import_modules.py`: Instead of importing ALL python modules in one UDF, fetch first list of installed python packages, and then call UDF one by one. This is to avoid an out-of memory issue. For example, importing `sagemaker` already allocates ~150MB on the heap.

### CODE REVIEW

**Review by kaklakariada — APPROVED** [review]

File: flavors/template-Exasol-all-python-3.8-cuda-conda/flavor_base/language_deps/packages/conda_packages [hunk 1]
```diff
@@ -1,7 +1,7 @@
 libxcrypt|4.4.36 # Needed for building the udfclient and because libcrypt got removed from conda sysroot, see https://github.com/conda-forge/linux-sysroot-feedstock/issues/52
 python|3.8.13
 numpy|1.24.3
-pandas|2.0.2
+pandas|2.0.3
```
Comment (tkilias) [inline flavors/template-Exasol-all-python-3.8-cuda-conda/flavor_base/language_deps/packages/conda_packages:4]: Why here Pandas 2.0.3 and in the normal template container 2.0.2

File: test_container/tests/test/standard-flavor/all/import_modules.py [hunk 2]
```diff
@@ -195,19 +224,23 @@ def run(ctx):
...
-        rows = self.query('''SELECT import_all_modules.import_all_modules() FROM dual''')
-        print("Number of modules:",len(rows))
-        failed_imports = [(row[0],row[1]) for row in rows if row[2] == "ERROR"]
-        for i in failed_imports:
-            print(i[0])
-        for i in failed_imports:
-            print(i[0], i[1])
-        self.assertEqual(failed_imports,[])
 
+    def test_import_all_modules(self):
```
Comment (tkilias) [inline test_container/tests/test/standard-flavor/all/import_modules.py:231]: Do you know what this costs us regarding test time. I assume the tests take several minutes longer now. However, we probably need to live with it.

**Review by tkilias — CHANGES_REQUESTED** [review]

File: test_container/tests/test/standard-flavor/all/import_modules.py (hunk 2 above)
Comment (tomuben) [reply test_container/tests/test/standard-flavor/all/import_modules.py:231]: When I run the test locally on my machine, it takes 1m7,650s. I think that's acceptable.

File: flavors/template-Exasol-all-python-3.8-cuda-conda/flavor_base/language_deps/packages/conda_packages (hunk 1 above)
Comment (tomuben) [reply flavors/template-Exasol-all-python-3.8-cuda-conda/flavor_base/language_deps/packages/conda_packages:4]: Update pandas also on the normal template flavor.

**Review by tkilias — APPROVED** [review]

**Pull request merged**